README.md

```diff
@@ -899,6 +900,61 @@ and need to be removed by hand.
 For long term use, a dedicated project in the Google Cloud Platform is
 recommended to prevent automated manipulation of important machines.
 
+<a name="openstack"/>
+
+## Openstack backend
+
+The Openstack backend is easy to setup and use, and allows distributing
+your tasks to an openstack environment.
+
+_$PROJECT/spread.yaml_
+```
+(...)
+
+backends:
+    openstack:
+        plan: cpu2-ram4-disk10
+        halt-timeout: 2h
+        systems:
+            - ubuntu-20.04:
+                  image: ubuntu-focal-daily-amd64
+                  workers: 5
+
+            # Extended syntax:
+            - another-system:
+                image: some-other-image
+
+```
+
+The Openstack backend gets all the information to authenticate from the
+environment variables. The following variables have to be set:
+```
+OS_AUTH_URL
+OS_PROJECT_ID
+OS_PROJECT_NAME
+OS_USER_DOMAIN_NAME
+OS_PROJECT_DOMAIN_ID (optional)
+OS_TENANT_ID
+OS_TENANT_NAME
+OS_USERNAME
+OS_PASSWORD
+OS_REGION_NAME
+OS_INTERFACE
+OS_IDENTITY_API_VERSION
+```
+
+You can set up those variables by sourcing the Openstack RC file. 
+For more information about the environment setup please read 
+https://docs.openstack.org/ocata/user-guide/common/cli-set-environment-variables-using-openstack-rc.html
+
+Images are located by first attempting to match the provided value exactly
+against the image name, if there is not exact match the most recent image
```

Comment:
thanks for the review @mvo5. In gce the images have associated a name, a family of images and a project, but in openstack the images just have a name. So it is not possible to follow the same mechanism that is used in gce. 

examples:
 openstack: https://paste.ubuntu.com/p/GjHjJZGcyH/
 gce: https://paste.ubuntu.com/p/k4npC2H4kF/

Reply:
Can I get a listing of what the "openstack image list" for the spread environment that we use for testing looks like? My worry is a bit that right now the code says: `strings.Contains(i.Name, imageName)` which feels a bit too lose for my taste, I would love to make it a bit stricter by having common prefixes/suffies or some regexp like way to be more certain that the image selected is actually the right image :)

Reply:
Hi, thanks for checking this one again.

ps5: https://paste.ubuntu.com/p/4vfF3JJwQ9/
canonistack: https://paste.ubuntu.com/p/pwmNVXXGsC/

Just consider that these images lists follow the convention defined by the owner, but other openstack providers could use a different naming convention. So, if in a different environment the images name don't follow the pattern prefixes/suffies, it will be impossible to boot those images.

Reply:
Thanks for these examples. So it seems like (in general) we cannot force the openstack instance to follow a convention we prescribe. If so, maybe we can be more precise. It seems there are two general cases:
1. The name is slightly weird/long like `auto-sync/ubuntu-bionic-18.04-amd64.img`. This should be fine and the user just needs to jut the long name in.
2. The name requires sorting (like `ubuntu/ubuntu-bionic-18.04-amd64-server-20220302-disk1.img`). But given that each image has a "created" property we can simply sort by that and use the latest image. So a strings.Contains() seems okay here. We could require a regex instead but it seems for the examples the `strings.Contains` plus sorting by creation date is sufficient. Wdyt @niemeyer ?

Reply:
This is done now, the image selection is now following what google does as much as possible, i.e. exact match wins always and if not the same "term" based matching as google is used. It's a bit more fuzzy/flexible compared to substring matching, I added a test to show this. I think this is what Gustavo expects but I will double check with him to be certain :)